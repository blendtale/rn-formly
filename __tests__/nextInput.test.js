import React from "react";
import { render } from "react-native-testing-library";
import Otp from  "../formComponent/otp";
import {
    StyleSheet,
    Dimensions
} from 'react-native'

const styles = StyleSheet.create({
    nextInputStyle: {
        textAlign: 'center', 
        marginLeft: 5, 
        marginRight: 5, 
        fontSize: 20,
        borderBottomWidth: 1,
        marginTop: 50
    }
})

describe('render otp component correct', () => {
 it('should render text like otp', () => {
    const props = {
        noOfTextInput:4,
        upsideEmit:() => {},
        value:[1,2,3,4],
        textInputStyle: [{width: (Dimensions.get('window').width * 0.6)/8 }, styles.nextInputStyle]
    }
    const component = render(<Otp {...props} />)
    // if component is getting rendered
    expect(component.toJSON()).toMatchSnapshot();
    // this checks if display value is there
    expect(component.getByDisplayValue("1")).toBeDefined();
    expect(component.getByDisplayValue("2")).toBeDefined();
    expect(component.getByDisplayValue("3")).toBeDefined();
    expect(component.getByDisplayValue("4")).toBeDefined();
 })
})
