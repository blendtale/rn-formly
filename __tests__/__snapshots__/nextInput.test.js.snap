// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render otp component correct should render text like otp 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    maxLength={1}
    numberOfLines={1}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "color": "black",
          "fontSize": 16,
          "marginLeft": 5,
          "marginRight": 5,
          "textAlign": "center",
        },
        Array [
          Object {
            "width": 56.25,
          },
          Object {
            "borderBottomWidth": 1,
            "fontSize": 20,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 50,
            "textAlign": "center",
          },
        ],
      ]
    }
    underlineColorAndroid="transparent"
    value="1"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    maxLength={1}
    numberOfLines={1}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "color": "black",
          "fontSize": 16,
          "marginLeft": 5,
          "marginRight": 5,
          "textAlign": "center",
        },
        Array [
          Object {
            "width": 56.25,
          },
          Object {
            "borderBottomWidth": 1,
            "fontSize": 20,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 50,
            "textAlign": "center",
          },
        ],
      ]
    }
    underlineColorAndroid="transparent"
    value="2"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    maxLength={1}
    numberOfLines={1}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "color": "black",
          "fontSize": 16,
          "marginLeft": 5,
          "marginRight": 5,
          "textAlign": "center",
        },
        Array [
          Object {
            "width": 56.25,
          },
          Object {
            "borderBottomWidth": 1,
            "fontSize": 20,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 50,
            "textAlign": "center",
          },
        ],
      ]
    }
    underlineColorAndroid="transparent"
    value="3"
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    maxLength={1}
    numberOfLines={1}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "color": "black",
          "fontSize": 16,
          "marginLeft": 5,
          "marginRight": 5,
          "textAlign": "center",
        },
        Array [
          Object {
            "width": 56.25,
          },
          Object {
            "borderBottomWidth": 1,
            "fontSize": 20,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 50,
            "textAlign": "center",
          },
        ],
      ]
    }
    underlineColorAndroid="transparent"
    value="4"
  />
</View>
`;
